src/main/java/frc/robot/commands/ChargeAutoCommand.java

```diff
@@ -1,14 +1,16 @@
 package frc.robot.commands;
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj.command.WaitCommand;
 import frc.robot.subsystems.DriveSubsystem;
 
-public class ChargeAutoCommand extends CommandBase {
+public class ChargeAutoCommand extends WaitCommand {
```

Comment:
I think `TimedCommand` is what you want? Unless `timeout` is the wrong word. What behavior are you looking or?

Reply:
TimedCommand isn't in wpilibNewCommands but WaitCommand is, so I assume WaitCommand is the newer version?

Reply:
Yeah you're right. It looks like `Command::withTimeout` is a thing so I would use that. But what behavior are you looking for? WaitCommand will delay the charge, whereas withTimeout will stop it after 1 second

Reply:
This seems to work after testing on the robot. I think the WaitCommand class solely defines a command that ends after a specified time, so when we extend it and add a execute method, it can run that method like a normal command and still end after a timeout.

Reply:
I would still recommend creating a command grouip that has a waitcommand, and also changing that param name to `delay` rather than `timeout` because it's not a timeout, it's a delay

Reply:
I don't think the command is meant to be subclassed in this way

Reply:
This seems to act as a delay when tested on the robot, but I agree that the class might not be subclassed like this usually. That being said, I think that the timeout should be applied to the `ChargeAutoCommand` through the constructor, rather than through `withTimeout` since a timeout is necessary for any usable charge auto.

Another option would be to extend `ParallelRaceGroup` and add a `WaitCommand` and a command that does the raw charge movement, but something along those lines seems overkill. Or we could just use `withTimeout`.

Reply:
I stand corrected! The official docs say that extending WaitCommand is a 'proper'. I also don't know why I didn't understand before but I think how you have it now actually is simple. I didn't realize there wasn't an `isFinished` command defined in this class. Now I understand you're using the `WaitCommand`'s `isFinished` method, so I like this.